science/config.py

```diff
@@ -3,18 +3,22 @@
 
 from __future__ import annotations
 
+import dataclasses
+import difflib
 import tomllib
-from collections import OrderedDict
+from collections import OrderedDict, defaultdict
 from dataclasses import dataclass
+from functools import cache
 from io import BytesIO
 from pathlib import Path
 from textwrap import dedent
-from typing import BinaryIO
+from typing import BinaryIO, DefaultDict, Generic, Iterator, Mapping, TypeVar
```

Comment:
AIUI from https://docs.python.org/3/library/typing.html#deprecated-aliases I think `defaultdict[K, V]`  works as a type annotation in 3.9+, and this code uses 3.11. Thus, potentially, this could be very slightly briefer:

```suggestion
from typing import BinaryIO, Generic, Iterator, Mapping, TypeVar
```

Reading that now makes me realise that this technically applies to `Iterator` and `Mapping` too (can use the "real" ones from `collections.abc`), which would be a big habit adjustment for me!

Reply:
Yeah, fixed + killed use of `OrderedDict` since dict has the insertion order guaranty I wanted now.